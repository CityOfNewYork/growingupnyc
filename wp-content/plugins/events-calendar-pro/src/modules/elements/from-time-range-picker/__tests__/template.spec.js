import React from 'react';
import renderer from 'react-test-renderer';
import FromTimeRangePicker from '../template';

describe( 'FromDateTimePicker', () => {
	test( 'should render component', () => {
		const component = renderer.create(
			<FromTimeRangePicker
				className="test-class"
				endTime="07:34"
				isAllDay={ false }
				isMultiDay={ false }
				onEndTimeChange={ jest.fn() }
				onEndTimeClick={ jest.fn() }
				onMultiDayChange={ jest.fn() }
				onStartTimeChange={ jest.fn() }
				onStartTimeClick={ jest.fn() }
				startTime="15:40"
			/>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	test( 'should render component with multi day', () => {
		const component = renderer.create(
			<FromTimeRangePicker
				className="test-class"
				endTime="07:34"
				isAllDay={ false }
				isMultiDay={ true }
				onEndTimeChange={ jest.fn() }
				onEndTimeClick={ jest.fn() }
				onMultiDayChange={ jest.fn() }
				onStartTimeChange={ jest.fn() }
				onStartTimeClick={ jest.fn() }
				startTime="15:40"
			/>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	test( 'should render component with all day', () => {
		const component = renderer.create(
			<FromTimeRangePicker
				className="test-class"
				endTime="07:34"
				isAllDay={ true }
				isMultiDay={ false }
				onEndTimeChange={ jest.fn() }
				onEndTimeClick={ jest.fn() }
				onMultiDayChange={ jest.fn() }
				onStartTimeChange={ jest.fn() }
				onStartTimeClick={ jest.fn() }
				startTime="15:40"
			/>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );
} );
