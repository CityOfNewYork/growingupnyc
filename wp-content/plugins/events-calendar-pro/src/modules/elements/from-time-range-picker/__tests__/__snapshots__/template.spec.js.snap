// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FromDateTimePicker should render component 1`] = `
<div
  className="tribe-editor__events-pro__row tribe-editor__labeled-row tribe-editor__from-time-range-picker test-class"
>
  <div
    className="tribe-editor__events-pro__label tribe-editor__labeled-row__label"
  >
    <span
      className="tribe-editor__events-pro__label-text"
    >
      From
    </span>
  </div>
  <div
    className="tribe-editor__labeled-row__content"
  >
    <div
      className="tribe-editor__timepicker"
    >
      <span>
        Dropdown
      </span>
    </div>
    <span>
      to
    </span>
    <div
      className="tribe-editor__timepicker"
    >
      <span>
        Dropdown
      </span>
    </div>
    <div
      className="tribe-editor__checkbox tribe-editor__multi-day-checkbox tribe-editor__from-time-range-picker__multi-day-checkbox"
    >
      <input
        checked={false}
        className="tribe-editor__input tribe-editor__input--checkbox tribe-editor__checkbox__input"
        id="uniqid"
        onChange={[MockFunction]}
        type="checkbox"
      />
      <label
        className="tribe-editor__checkbox__label"
        htmlFor="uniqid"
      >
        Multi-day
      </label>
    </div>
  </div>
</div>
`;

exports[`FromDateTimePicker should render component with all day 1`] = `
<div
  className="tribe-editor__events-pro__row tribe-editor__labeled-row tribe-editor__from-time-range-picker test-class"
>
  <div
    className="tribe-editor__events-pro__label tribe-editor__labeled-row__label"
  >
    <span
      className="tribe-editor__events-pro__label-text"
    >
      From
    </span>
  </div>
  <div
    className="tribe-editor__labeled-row__content"
  >
    <div
      className="tribe-editor__timepicker"
    >
      <span>
        Dropdown
      </span>
    </div>
    <div
      className="tribe-editor__checkbox tribe-editor__multi-day-checkbox tribe-editor__from-time-range-picker__multi-day-checkbox"
    >
      <input
        checked={false}
        className="tribe-editor__input tribe-editor__input--checkbox tribe-editor__checkbox__input"
        id="uniqid"
        onChange={[MockFunction]}
        type="checkbox"
      />
      <label
        className="tribe-editor__checkbox__label"
        htmlFor="uniqid"
      >
        Multi-day
      </label>
    </div>
  </div>
</div>
`;

exports[`FromDateTimePicker should render component with multi day 1`] = `
<div
  className="tribe-editor__events-pro__row tribe-editor__labeled-row tribe-editor__from-time-range-picker tribe-editor__from-time-range-picker--multi-day test-class"
>
  <div
    className="tribe-editor__events-pro__label tribe-editor__labeled-row__label"
  >
    <span
      className="tribe-editor__events-pro__label-text"
    >
      From
    </span>
  </div>
  <div
    className="tribe-editor__labeled-row__content"
  >
    <div
      className="tribe-editor__timepicker"
    >
      <span>
        Dropdown
      </span>
    </div>
    <span>
      to
    </span>
    <div
      className="tribe-editor__timepicker"
    >
      <span>
        Dropdown
      </span>
    </div>
    <div
      className="tribe-editor__checkbox tribe-editor__multi-day-checkbox tribe-editor__from-time-range-picker__multi-day-checkbox"
    >
      <input
        checked={true}
        className="tribe-editor__input tribe-editor__input--checkbox tribe-editor__checkbox__input"
        id="uniqid"
        onChange={[MockFunction]}
        type="checkbox"
      />
      <label
        className="tribe-editor__checkbox__label"
        htmlFor="uniqid"
      >
        Multi-day
      </label>
    </div>
  </div>
</div>
`;
